jenkinsfile.groovy

```diff
@@ -257,6 +267,52 @@ pipeline {
         } //stage
       } //parallel
     } //stage
+    stage ("Tests") {
+      agent { node { label "${jenkinsCtrlNodeLabel} && ${account}" } }
+      when  { expression { params.action == 'apply' }}
+      steps {
+        script {
+          repoFunctionsFactory.checkoutGitRepo(github.cvr_app.url, params.branch, github.cvr_app.name, globalValuesFactory.SSH_DEPLOY_GIT_CREDS_ID)
+
+          dir(github.cvr_app.name + "/selenium") {
+            Integer testStatus = sh(
+                    returnStatus: true,
+                    script: """
+                        mkdir -p ./${seleniumScreenshotsDir}
+
+                        ./gradlew clean build \
+                          -Dtest.baseUrl=${recallsApiGwUrl}\
+                          -Dtest.platform=linux \
+                          -Dtest.browserName=firefox \
+                          -Dtest.gridEnabled=selenium \
+                          -Dtest.javascript.enabled=yes \
+                          -Dtest.gridUrl=http://${params.selenium_hub_address}/wd/hub/ \
+                          -Dtest.screenshots.error.enabled=yes \
+                          -Dtest.screenshots.error.folder=./${seleniumScreenshotsDir}
+                      """)
+
+            if (testStatus != 0) {
+
+              archiveArtifacts(
+                      artifacts: "${seleniumScreenshotsDir}/**",
+                      onlyIfSuccessful: false,
+                      allowEmptyArchive: true,
+              )
+
+              publishHTML target: [
+                      allowMissing: false,
+                      reportDir: "./build/reports/tests/selenium",
+                      reportFiles: 'index.html',
+                      reportName: 'Selenium Report'
+              ]
+
+              failure("Selenium failure. Status code:" + testStatus)
```

Comment:
you can do "Selenium failure. Status code: ${testStatus}" in groovy, looks better and its easier to see whitspaces in the string